client/service/app.js

```diff
@@ -0,0 +1,335 @@
+import { Deck } from './models/Deck.js';
+import 'dotenv/config'
+
+const app = express()
+const port = 8000
+// Connect to MongoDB
+const connectionString = `mongodb+srv://${process.env.DB_USER}:${process.env.DB_PASSWORD}@cluster0.bgjij.mongodb.net/myFirstDatabase?retryWrites=true&w=majority`
+try {
+  await mongoose.connect(connectionString, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    serverSelectionTimeoutMS: 5000})
+  console.log("Connected to the database!");
+} catch (err) {
+  console.log("Cannot connect to the database!", err);
+  process.exit();
+}
+/* // Middleware
+const exampleMiddleware = (req, res, next) => {
+  console.log('example middleware')
+  next()
+} */
+
+app.use(cors());
+app.use(bodyParser.json());
+app.use(express.json());
+app.use(bodyParser.urlencoded({extended: true}));
+
+// --------------------------------------------------------------------------------
+// ---------------- ROUTES --------------------------------------------------------
+// --------------------------------------------------------------------------------
+
+// ----------------------------------------
+// ------------- USER ROUTES --------------
+// ----------------------------------------
+
+// ---------------- Get all users ----------------
+app.get('/users', async (req, res) => { 
+  const users = await Users.find();
+  res.json(users);
+});
+
+// ---------------- Create a user ----------------
+app.post('/createuser', async (req, res) => {
+  const userData = new Users(req.body);
+  if (userData.id === undefined) {
+      userData.id = mongoose.Types.ObjectId();
+  }   
+  try {
+      const user = await Users.create(userData);
+      res.status(200);
+      res.json(user);
+  } 
+  catch (error) {
+      res.status(400);
+      res.send(error);
+  }   
+});
+
+// ---------------- Get A user ----------------
+app.get('/get_user/:id', async (req, res) => {
+  if (req.params.id === undefined) {
+      res.sendStatus(400);
+  } else {
+      const id = req.params.id;
+      const user = await Users.findById(id);
+      if (user === null) {
+          res.sendStatus(404);
+      } else {
+          res.status(200);
+          res.json(user);
+      }
+  }
+});
+
+// ---------------- Update a user ----------------
+app.post('/update_user/:id', async (req, res) => {
+  if (req.params.id === undefined) {
+      res.status(400);
+      res.send('No id provided');
+  }
+  else {
+    const user = await Users.findById(req.params.id);
+    if (user === null) {
+        res.status(404);
+        res.send('User not found');
+    } 
+    else {
+      for (const key in req.body) {
+          user[key] = req.body[key];
+      }
+      try {
+        await user.save();
+        res.status(200);
+        res.json(user);
+      }
+      catch (error) {
+        res.status(400);
+        res.send(error);
+      }
+  }
+}
+});
+
+// ---------------- Delete a user ----------------
+app.get('/delete_user/:id', async (req, res) => {
```

Comment:
Remember that the http method should correspond to the operation. You *can* use get here, but you should use delete for delete operations. 
This also means you don't need the word 'delete' or the underscore there. RESTful APIs focus on using the HTTP method to describe the operation and nouns in the path to describe the resource. 
```suggestion
app.delete('/user/:id', async (req, res) => {
```